client.getOnlySource(SourceType.HDFS) call failed with stack trace

I have the following code snippet to retrieve the HDFS source type. When executing my program in the Cloudera AWS clusters, I got the following stack trace.

        NavApiCient client = plugin.getClient();
        Source hdfsSource = client.getOnlySource(SourceType.HDFS);
  
============
16:31:08.006 [main] INFO  c.w.i.cloudera.SensitiveData - Adapter started.
Exception in thread "main" org.springframework.http.converter.HttpMessageNotReadableException: Could not read JSON: Numeric value (2160620923) out of range of int
 at [Source: org.apache.http.client.entity.LazyDecompressingInputStream@cad7e2b; line: 17, column: 40] (through reference chain: com.cloudera.nav.sdk.client.SourceAttrs["sourceExtractIteration"]); nested exception is com.fasterxml.jackson.databind.JsonMappingException: Numeric value (2160620923) out of range of int
 at [Source: org.apache.http.client.entity.LazyDecompressingInputStream@cad7e2b; line: 17, column: 40] (through reference chain: com.cloudera.nav.sdk.client.SourceAttrs["sourceExtractIteration"])
	at org.springframework.http.converter.json.MappingJackson2HttpMessageConverter.readJavaType(MappingJackson2HttpMessageConverter.java:228)
	at org.springframework.http.converter.json.MappingJackson2HttpMessageConverter.read(MappingJackson2HttpMessageConverter.java:220)
	at org.springframework.web.client.HttpMessageConverterExtractor.extractData(HttpMessageConverterExtractor.java:95)
	at org.springframework.web.client.RestTemplate$ResponseEntityResponseExtractor.extractData(RestTemplate.java:795)
	at org.springframework.web.client.RestTemplate$ResponseEntityResponseExtractor.extractData(RestTemplate.java:779)
	at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:559)
	at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:512)
	at org.springframework.web.client.RestTemplate.exchange(RestTemplate.java:454)
	at com.cloudera.nav.sdk.client.NavApiCient.getAllSources(NavApiCient.java:82)
	at com.cloudera.nav.sdk.client.NavApiCient.loadAllSources(NavApiCient.java:220)
	at com.cloudera.nav.sdk.client.NavApiCient.getSourcesForType(NavApiCient.java:172)
	at com.cloudera.nav.sdk.client.NavApiCient.getOnlySource(NavApiCient.java:163)
	at com.waterlinedata.integration.cloudera.SensitiveData.Main(SensitiveData.java:154)
	at com.waterlinedata.integration.cloudera.SensitiveData.main(SensitiveData.java:237)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Numeric value (2160620923) out of range of int
 at [Source: org.apache.http.client.entity.LazyDecompressingInputStream@cad7e2b; line: 17, column: 40] (through reference chain: com.cloudera.nav.sdk.client.SourceAttrs["sourceExtractIteration"])
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:232)
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:197)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.wrapAndThrow(BeanDeserializerBase.java:1311)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:297)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:121)
	at com.fasterxml.jackson.databind.deser.std.ObjectArrayDeserializer.deserialize(ObjectArrayDeserializer.java:151)
	at com.fasterxml.jackson.databind.deser.std.ObjectArrayDeserializer.deserialize(ObjectArrayDeserializer.java:18)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:2888)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2094)
	at org.springframework.http.converter.json.MappingJackson2HttpMessageConverter.readJavaType(MappingJackson2HttpMessageConverter.java:225)
	... 13 more
Caused by: com.fasterxml.jackson.core.JsonParseException: Numeric value (2160620923) out of range of int
 at [Source: org.apache.http.client.entity.LazyDecompressingInputStream@cad7e2b; line: 17, column: 40]
	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1369)
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:532)
	at com.fasterxml.jackson.core.base.ParserBase.convertNumberToInt(ParserBase.java:847)
	at com.fasterxml.jackson.core.base.ParserBase.getIntValue(ParserBase.java:643)
	at com.fasterxml.jackson.databind.deser.std.StdDeserializer._parseInteger(StdDeserializer.java:306)
	at com.fasterxml.jackson.databind.deser.std.NumberDeserializers$IntegerDeserializer.deserialize(NumberDeserializers.java:319)
	at com.fasterxml.jackson.databind.deser.std.NumberDeserializers$IntegerDeserializer.deserialize(NumberDeserializers.java:301)
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:464)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:98)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:295)
	... 19 more
[waterline@ip-192-168-100-254 bin]$

After changing the call and it still crashes. See the new stack trace below. You can see that I have replaced the call with the following:

      Collection<Source> hdfsSources = client.getSourcesForType(SourceType.HDFS);
=================
Exception in thread "main" org.springframework.http.converter.HttpMessageNotReadableException: Could not read JSON: Numeric value (2978969079) out of range of int
 at [Source: org.apache.http.client.entity.LazyDecompressingInputStream@6848db81; line: 17, column: 40] (through reference chain: com.cloudera.nav.sdk.client.SourceAttrs["sourceExtractIteration"]); nested exception is com.fasterxml.jackson.databind.JsonMappingException: Numeric value (2978969079) out of range of int
 at [Source: org.apache.http.client.entity.LazyDecompressingInputStream@6848db81; line: 17, column: 40] (through reference chain: com.cloudera.nav.sdk.client.SourceAttrs["sourceExtractIteration"])
	at org.springframework.http.converter.json.MappingJackson2HttpMessageConverter.readJavaType(MappingJackson2HttpMessageConverter.java:228)
	at org.springframework.http.converter.json.MappingJackson2HttpMessageConverter.read(MappingJackson2HttpMessageConverter.java:220)
	at org.springframework.web.client.HttpMessageConverterExtractor.extractData(HttpMessageConverterExtractor.java:95)
	at org.springframework.web.client.RestTemplate$ResponseEntityResponseExtractor.extractData(RestTemplate.java:795)
	at org.springframework.web.client.RestTemplate$ResponseEntityResponseExtractor.extractData(RestTemplate.java:779)
	at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:559)
	at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:512)
	at org.springframework.web.client.RestTemplate.exchange(RestTemplate.java:454)
	at com.cloudera.nav.sdk.client.NavApiCient.getAllSources(NavApiCient.java:82)
	at com.cloudera.nav.sdk.client.NavApiCient.loadAllSources(NavApiCient.java:220)
	at com.cloudera.nav.sdk.client.NavApiCient.getSourcesForType(NavApiCient.java:172)
	at com.waterlinedata.integration.cloudera.SensitiveData.Main(SensitiveData.java:207)
	at com.waterlinedata.integration.cloudera.SensitiveData.main(SensitiveData.java:264)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Numeric value (2978969079) out of range of int
 at [Source: org.apache.http.client.entity.LazyDecompressingInputStream@6848db81; line: 17, column: 40] (through reference chain: com.cloudera.nav.sdk.client.SourceAttrs["sourceExtractIteration"])
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:232)
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:197)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.wrapAndThrow(BeanDeserializerBase.java:1311)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:297)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:121)
	at com.fasterxml.jackson.databind.deser.std.ObjectArrayDeserializer.deserialize(ObjectArrayDeserializer.java:151)
	at com.fasterxml.jackson.databind.deser.std.ObjectArrayDeserializer.deserialize(ObjectArrayDeserializer.java:18)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:2888)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2094)
	at org.springframework.http.converter.json.MappingJackson2HttpMessageConverter.readJavaType(MappingJackson2HttpMessageConverter.java:225)
	... 12 more
Caused by: com.fasterxml.jackson.core.JsonParseException: Numeric value (2978969079) out of range of int
 at [Source: org.apache.http.client.entity.LazyDecompressingInputStream@6848db81; line: 17, column: 40]
	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1369)
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:532)
	at com.fasterxml.jackson.core.base.ParserBase.convertNumberToInt(ParserBase.java:847)
	at com.fasterxml.jackson.core.base.ParserBase.getIntValue(ParserBase.java:643)
	at com.fasterxml.jackson.databind.deser.std.StdDeserializer._parseInteger(StdDeserializer.java:306)
	at com.fasterxml.jackson.databind.deser.std.NumberDeserializers$IntegerDeserializer.deserialize(NumberDeserializers.java:319)
	at com.fasterxml.jackson.databind.deser.std.NumberDeserializers$IntegerDeserializer.deserialize(NumberDeserializers.java:301)
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:464)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:98)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:295)
	... 18 more

--- a/client/src/main/java/com/cloudera/nav/sdk/client/MetadataExtractor.java
+++ b/client/src/main/java/com/cloudera/nav/sdk/client/MetadataExtractor.java
@@ -104,14 +104,14 @@ public class MetadataExtractor {
                                            String entitiesQuery,
                                            String relationsQuery) {
     try {
-      TypeReference<Map<String, Integer>> typeRef =
-          new TypeReference<Map<String, Integer>>(){};
+      TypeReference<Map<String, Long>> typeRef =
+          new TypeReference<Map<String, Long>>(){};
       Iterable<String> extractorQuery;
-      Map<String, Integer> endMarker;
+      Map<String, Long> endMarker;
       if(StringUtils.isEmpty(startMarkerRep) && StringUtils.isEmpty(endMarkerRep)){
         extractorQuery = Lists.newArrayList("*");
       } else {
-        Map<String, Integer> startMarker;
+        Map<String, Long> startMarker;
         if(StringUtils.isEmpty(startMarkerRep)) {
           startMarker = getNavMarker(false);
         } else {
@@ -137,12 +137,12 @@ public class MetadataExtractor {
    *
    * @return Map of sourceId to its to extractIteration
    */
-  private Map<String, Integer> getNavMarker(boolean current) {
+  private Map<String, Long> getNavMarker(boolean current) {
     Collection<Source> sources = client.getAllSources();
-    HashMap<String, Integer> newMarker = Maps. newHashMap();
+    HashMap<String, Long> newMarker = Maps. newHashMap();
     for (Source source : sources) {
       String id = source.getIdentity();
-      Integer sourceExtractIteration = (current) ?
+      Long sourceExtractIteration = (current) ?
           source.getSourceExtractIteration() : 0;
       newMarker.put(id, sourceExtractIteration);
     }
@@ -157,12 +157,12 @@ public class MetadataExtractor {
    * @param m2 Marker for later(current) extraction state
    * @return Iterable of possible extractorRunIds to be used in queries
    */
-  private Iterable<String> getExtractorQueryList(Map<String, Integer> m1,
-                                         Map<String, Integer> m2) {
+  private Iterable<String> getExtractorQueryList(Map<String, Long> m1,
+                                                 Map<String, Long> m2) {
     List<String> runIdList= Lists.newArrayList();
     for (String key: m1.keySet()) {
-      for (int i=m1.get(key); i<(m2.get(key)+1); i++){
-        String possible = key + "##" + Integer.toString(i);
+      for (long i=m1.get(key); i<(m2.get(key)+1); i++) {
+        String possible = key + "##" + Long.toString(i);
         runIdList.add(possible);
       }
     }
@@ -198,7 +198,7 @@ public class MetadataExtractor {
    * @return String representation of a marker
    */
   public String getMarker() {
-    Map<String, Integer> currentMarker = getNavMarker(true);
+    Map<String, Long> currentMarker = getNavMarker(true);
     try {
       return new ObjectMapper().writeValueAsString(currentMarker);
     } catch (IOException e){

--- a/client/src/main/java/com/cloudera/nav/sdk/client/SourceAttrs.java
+++ b/client/src/main/java/com/cloudera/nav/sdk/client/SourceAttrs.java
@@ -31,7 +31,7 @@ class SourceAttrs {
   private SourceType sourceType;
   private String sourceUrl;
   private String identity;
-  private Integer sourceExtractIteration;
+  private Long sourceExtractIteration;
 
   public String getClusterName() {
     return clusterName;
@@ -73,11 +73,11 @@ class SourceAttrs {
     this.sourceType = sourceType;
   }
 
-  public Integer getSourceExtractIteration() {
+  public Long getSourceExtractIteration() {
     return sourceExtractIteration;
   }
 
-  public void setSourceExtractIteration(Integer sourceExtractIteration) {
+  public void setSourceExtractIteration(Long sourceExtractIteration) {
     this.sourceExtractIteration = sourceExtractIteration;
   }
 

--- a/client/src/test/java/com/cloudera/nav/sdk/client/JsonMetadataWriterTest.java
+++ b/client/src/test/java/com/cloudera/nav/sdk/client/JsonMetadataWriterTest.java
@@ -67,7 +67,7 @@ public class JsonMetadataWriterTest {
   @Test
   public void testWriteEntity() throws IOException {
     Source source = new Source("HDFS-1", SourceType.HDFS, "Cluster",
-        "http://ns1", 0);
+        "http://ns1", 0L);
     HdfsEntity entity = new HdfsEntity();
     entity.setSourceId(source.getIdentity());
     entity.setFileSystemPath("/user/test");
@@ -110,7 +110,7 @@ public class JsonMetadataWriterTest {
   @Test
   public void testChangeSets() throws IOException {
     Source source = new Source("HDFS-1", SourceType.HDFS, "Cluster",
-        "http://ns1", 1);
+        "http://ns1", 1L);
     HdfsEntity entity = new HdfsEntity("/user/test",
         EntityType.DIRECTORY, source.getIdentity());
     entity.setTags("foo", "bar");
@@ -159,7 +159,7 @@ public class JsonMetadataWriterTest {
   @Test
   public void testWriteRelation() throws IOException {
     Source source = new Source("HDFS-1", SourceType.HDFS, "Cluster",
-        "http://ns1", 1);
+        "http://ns1", 1L);
     HdfsEntity inputData = new HdfsEntity("/user/test/input",
         EntityType.DIRECTORY, source.getIdentity());
     inputData.setTags(ImmutableList.of("foo", "bar"));
@@ -206,7 +206,7 @@ public class JsonMetadataWriterTest {
   @Test
   public void testWriteComposite() throws IOException {
     Source source = new Source("ExternalApp", SourceType.SDK, "ExternalApp",
-        "http://appHost:port", 0);
+        "http://appHost:port", 0L);
     CustomOperationExecution exec = prepExec(source);
     JsonMetadataWriter mWriter = new JsonMetadataWriter(config, stream,
         mockConn);

--- a/client/src/test/java/com/cloudera/nav/sdk/client/MetadataExtractorTest.java
+++ b/client/src/test/java/com/cloudera/nav/sdk/client/MetadataExtractorTest.java
@@ -44,7 +44,7 @@ public class MetadataExtractorTest {
     extractor = new MetadataExtractor(client, null);
     marker1Rep = "{\"identityString\":100}";
     Source source1 = new Source("source1", SourceType.HDFS, "cluster1",
-        "foo/bar", "identityString", 100);
+        "foo/bar", "identityString", 100L);
     when(client.getAllSources()).thenReturn(Lists.newArrayList(source1));
   }
 

--- a/model/src/main/java/com/cloudera/nav/sdk/model/Source.java
+++ b/model/src/main/java/com/cloudera/nav/sdk/model/Source.java
@@ -28,7 +28,7 @@ public class Source {
   private final String clusterName;
   private final String sourceUrl;
   private final String identity;
-  private final Integer sourceExtractIteration;
+  private final Long sourceExtractIteration;
 
 
   /**
@@ -39,7 +39,7 @@ public class Source {
    */
   public Source(String name, SourceType sourceType,
                 String clusterName, String sourceUrl,
-                String identity, Integer sourceExtractIteration) {
+                String identity, Long sourceExtractIteration) {
     Preconditions.checkNotNull(identity);
     this.name = name;
     this.sourceType = sourceType;
@@ -50,7 +50,7 @@ public class Source {
   }
 
   public Source(String name, SourceType sourceType,
-                String clusterName, String sourceUrl, Integer sourceExtractIteration) {
+                String clusterName, String sourceUrl, Long sourceExtractIteration) {
     this(name, sourceType, clusterName, sourceUrl,
         SourceIdGenerator.generateSourceId(clusterName, name), sourceExtractIteration);
   }
@@ -99,7 +99,7 @@ public class Source {
     return identity;
   }
 
-  public Integer getSourceExtractIteration(){ return sourceExtractIteration; }
+  public Long getSourceExtractIteration(){ return sourceExtractIteration; }
 
   @Override
   public boolean equals(Object o) {

--- a/model/src/test/java/com/cloudera/nav/sdk/model/SourceTest.java
+++ b/model/src/test/java/com/cloudera/nav/sdk/model/SourceTest.java
@@ -27,10 +27,10 @@ public class SourceTest {
     String sourceName = "mySource";
     String clusterName = "myCluster";
     Source src = new Source(sourceName, SourceType.NONE,
-        clusterName, "http://host:port", 5);
+        clusterName, "http://host:port", 5L);
     assertEquals(MD5IdGenerator.generateIdentity(clusterName, sourceName),
         src.getIdentity());
     assertEquals(src, new Source(sourceName, SourceType.HDFS,
-        clusterName, "http://newHost:newPort", 5));
+        clusterName, "http://newHost:newPort", 5L));
   }
 }

--- a/model/src/test/java/com/cloudera/nav/sdk/model/entities/EntityTest.java
+++ b/model/src/test/java/com/cloudera/nav/sdk/model/entities/EntityTest.java
@@ -29,7 +29,7 @@ public class EntityTest {
   @Test
   public void testHdfsEntity() {
     Source hdfs1 = new Source("HDFS-1", SourceType.HDFS, "Cluster",
-        "http://ns1",1);
+        "http://ns1", 1L);
 
     HdfsEntity entity = new HdfsEntity();
     entity.setEntityType(EntityType.DIRECTORY);
@@ -43,7 +43,7 @@ public class EntityTest {
   @Test
   public void testHiveTable() {
     Source hive1 = new Source("HIVE-1", SourceType.HIVE, "Cluster",
-        "http://hive-server:port", 1);
+        "http://hive-server:port", 1L);
     HiveTable entity = new HiveTable();
     entity.setDatabaseName("db");
     entity.setTableName("table");
@@ -57,7 +57,7 @@ public class EntityTest {
   @Test
   public void testHiveColumn() {
     Source hive1 = new Source("HIVE-1", SourceType.HIVE, "Cluster",
-        "http://hive-server:port",1);
+        "http://hive-server:port", 1L);
     HiveColumn entity = new HiveColumn();
     entity.setDatabaseName("db");
     entity.setTableName("table");

